conda_forge_tick/migrators.xsh

```diff
@@ -368,3 +373,88 @@ class Compiler(Migrator):
 
     def remote_branch(self):
         return 'compiler_migration2'
+
+
+class Noarch(Migrator):
+    """Migrator for adding noarch."""
+    migrator_version = 0
+
+    compiler_pat = re.compile('.*_compiler_stub')
+    sel_pat = re.compile('(.+?)\s*(#.*)?\[([^\[\]]+)\](?(2)[^\(\)]*)$')
+    unallowed_reqs = ['toolchain', 'gcc', 'cython', 'clangdev']
+
+    def filter(self, attrs):
+        conditional = (super().filter(attrs) or
+                       attrs.get('meta_yaml', {}).get('outputs') or
+                       attrs.get('meta_yaml', {}).get('build', {}).get('noarch')
+                      )
+        if conditional:
+            return True
+        python = False
+        for req in attrs.get('req', []):
+            if self.compiler_pat.match(req) or req in self.unallowed_reqs:
+                return True
+            if req == 'python':
+                python = True
+        if not python:
+            return True
+        for line in attrs.get('raw_meta_yaml', '').splitlines():
+            if self.sel_pat.match(line):
+                return True
+        return False
+
+    def migrate(self, recipe_dir, attrs, **kwargs):
+        with indir(recipe_dir):
+            replace_in_file(
+                    'build:',
+                    'build:\n  noarch: python',
+                    'meta.yaml',
+                    leading_whitespace=False)
+            replace_in_file(
+                    'script:.+?',
+                    'script: python -m pip install --no-deps --ignore-installed .',
+                    'meta.yaml')
+            if 'pip' not in attrs['req']:
+                replace_in_file(
+                        '  build:',
+                        '  build:\n    - pip',
+                        'meta.yaml',
+                        leading_whitespace=False)
+        return self.migrator_uid(attrs)
+
+    def pr_body(self):
+        body = super().pr_body()
+        body = body.format(
+                    'I think this feedstock should be built with noarch.\n'
+                    'This means that the package only needs to be built '
+                    'once, drastically reducing CI usage.\n'
+                    'See [here](https://conda-forge.org/docs/meta.html#building-noarch-packages)'
+                    'for more information about building noarch packages.'
+                    'Before merging this PR make sure:\n'
+                    '- [ ] No compiled extensions\n'
```

Comment:
I would do a list comprehension to build this, since we may need to add more at some point (and this might get externally managed?). (also this has a ton of `- [ ]` boiler plate)

Reply:
Additionally as we get better at inspecting these things we might be able to fill some in/hint that they are taken care of.